File: back/src/main/java/com/back/domain/queue/event/QueueStatusEvent.java
package com.back.domain.queue.event;

import java.time.LocalDateTime;

public record QueueStatusEvent(
        Long scheduleId,
        Long currentRank,
        Long totalWaitingCount,
        LocalDateTime updatedAt
) {
    public static QueueStatusEvent of(Long scheduleId, Long currentAllowedSequence, Long totalWaitingCount) {
        return new QueueStatusEvent(scheduleId, currentAllowedSequence, totalWaitingCount, LocalDateTime.now());
    }
}

File: back/src/main/kotlin/com/back/domain/queue/event/QueueStatusEvent.kt
package com.back.domain.queue.event

import java.time.LocalDateTime

data class QueueStatusEvent(
    val scheduleId: Long,
    val currentRank: Long,
    val totalWaitingCount: Long,
    val updatedAt: LocalDateTime
) {
    companion object {
        fun of(scheduleId: Long, currentAllowedSequence: Long, totalWaitingCount: Long): QueueStatusEvent {
            return QueueStatusEvent(
                scheduleId = scheduleId,
                currentRank = currentAllowedSequence,
                totalWaitingCount = totalWaitingCount,
                updatedAt = LocalDateTime.now()
            )
        }
    }
}
